students/Josh_HOff/lesson03/mailroom.py

```diff
@@ -0,0 +1,74 @@
+#email sent out to donor
+def email(donor, donation):
+    print('')
+    print(f'Hello, {donor}! Thank you very much for your generous donation of ${donation}! Your contribution is essential and will be well utilized.')
+
+#function if user chooses to send a thank you email
+def response1(donors_list):
+    while True:
+        x = 0
+        print('')
+        response = input('Type a Full Name: ')
+        if response == 'list':
+            for i in donors_list:
+                print(i[0])
+        elif response == 'quit':
+            return(donors_list)
+        else:
+            print('')
+            don_amt = input('What is the donation amount?: ')
+            if don_amt == 'quit':
+                return donors_list
+            else:
+                float(don_amt)
+                count = 0
+                for i in donors_list:
+                    if i[0] == response:
+                        donors_list[count].append(float(don_amt))
+                        email(response, don_amt)
+                        x = 2
+                        break
+                    count += 1
+                if x != 2:
```

Comment:
what is `x` and the significance of 2?

Reply:
here is an example of a more concise version using a list for a data structure:
```
def send_thank_you():
    while True:
        resp = input("Please type 'list', 'quit' or name of the donor")
        if resp == "list":
            list_donors()
        elif resp == "quit":
            return
        else:
            new_donation = input("Please enter amount")
            new_donor = False
            for donor_name, donations in donors_list:
                if donor_name == resp:
                    donations.append(new_donation)
                    break
            else:
                new_donor = True
                donors_list.append((resp, [new_donation]))
            
            print_email(new_donor)  # passed bool for custom message
            return
```

Reply:
I was trying to avoid executing this block of code if the previous block of code was executed, so I used the variable x.

>                 for i in donors_list:
>                     if i[0] == response:
>                         donors_list[count].append(float(don_amt))
>                         email(response, don_amt)
>                         x = 2
>                         break
>                     count += 1
>                 if x != 2:
>                     donors_list += ([response],)
>                     donors_list[count].append(float(don_amt))
>                     email(response, don_amt)
>                 return(donors_list)
>

Reply:
Your block of code makes a LOT more sense and is much cleaner. I'll put it in comments in my code for future reference.

Reply:
Nice, I understand now. If I need to use control flow like that in the future, I'll use boolean flags. I didn't have to use them in mailroom2.